src/leo_gateway_http_commons.erl

```diff
@@ -738,30 +738,63 @@ get_range_object(Req, Bucket, Key, {error, badarg}) ->
     ?reply_bad_range([?SERVER_HEADER], Key, <<>>, Req);
 get_range_object(Req, Bucket, Key, {_Unit, Range}) when is_list(Range) ->
     Mime = leo_mime:guess_mime(Key),
-    Header = [?SERVER_HEADER,
-              {?HTTP_HEAD_RESP_CONTENT_TYPE,  Mime}],
-    {ok, Req2} = cowboy_req:chunked_reply(?HTTP_ST_PARTIAL_CONTENT, Header, Req),
-    get_range_object_1(Req2, Bucket, Key, Range, undefined).
+    case get_body_length_1(Key, Range) of
+        error ->
+            ?reply_service_unavailable_error([?SERVER_HEADER], Key, <<>>, Req);
```

Comment:
To respond the http status code properly, 
We need to check why this error occured,  and respond http status corresponding with the error like

 - {error, not_found} reply_not_found_without_body
 - {error, unavailable} reply_service_unavailable_error
 - {error, ?ERR_TYPE_INTERNAL_ERROR} reply_internal_error_without_body
 - {error, timeout} reply_timeout_without_body